l3kernel/l3token.dtx

```diff
@@ -974,20 +904,35 @@
 %   appropriate to the result of the test).
 % \end{function}
 %
-% \begin{function}[updated = 2012-12-05, TF]
-%   {\peek_meaning_remove_ignore_spaces:N}
+% \begin{function}[added = 2018-10-01]{\peek_remove_spaces:n}
 %   \begin{syntax}
-%     \cs{peek_meaning_remove_ignore_spaces:NTF} \meta{test token}
-%     ~~\Arg{true code} \Arg{false code}
+%     \cs{peek_remove_spaces:n} \Arg{code}
 %   \end{syntax}
-%   Tests if the next non-space \meta{token} in the input stream has the
-%   same meaning as the \meta{test token} (as defined by the test
-%   \cs{token_if_eq_meaning:NNTF}).  Explicit and implicit space tokens
-%   (with character code 32 and category code 10) are ignored and
-%   removed by the test and the \meta{token} is removed from the
-%   input stream if the test is true. The function then places
-%   either the \meta{true code} or \meta{false code} in the input stream
-%   (as appropriate to the result of the test).
+%   Peeks ahead and detect if the following token is a space (category code
+%   $10$ and character code $32$). If so, removes the token and checks the
+%   next token. Once a non-space token is found, the \meta{code} will be
+%   inserted into the input stream. Typically this will contain a \texttt{peek}
+%   operation, but this is not required.
+% \end{function}
+%
+% \begin{function}[added = 2022-01-10]{\peek_remove_filler:n}
+%   \begin{syntax}
+%     \cs{peek_remove_filler:n} \Arg{code}
+%   \end{syntax}
+%   Peeks ahead and detect if the following token is a space (category code
+%   $10$ and character code $32$) or has meaning equal to \cs{scan_stop:}. If so,
```

Comment:
Here it should certainly be just a catcode test to match what TeX does (the code below is correct).  On the other hand `\peek_remove_spaces:n` should remain unchanged because the use case is to ignore actual spaces in the file.

Reply:
agreed